controllers/operatorconfigmap_controller.go

```diff
@@ -0,0 +1,454 @@
+	operatorConfigMap *corev1.ConfigMap
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (c *OperatorConfigMapReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	clusterVersionPredicates := builder.WithPredicates(
+		predicate.GenerationChangedPredicate{},
+	)
+
+	configMapPredicates := builder.WithPredicates(
+		predicate.NewPredicateFuncs(
+			func(client client.Object) bool {
+				namespace := client.GetNamespace()
+				name := client.GetName()
+				return ((namespace == c.OperatorNamespace) && (name == operatorConfigMapName))
+			},
+		),
+	)
+	// Reconcile the ClusterVersion object when the operator config map is updated
+	enqueueConfigMapRequest := handler.EnqueueRequestsFromMapFunc(
+		func(_ context.Context, client client.Object) []reconcile.Request {
+			return []reconcile.Request{{
+				NamespacedName: types.NamespacedName{
+					Name: operatorConfigMapName,
+				},
+			}}
+		},
+	)
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.ConfigMap{}, configMapPredicates).
+		Watches(&configv1.ClusterVersion{}, enqueueConfigMapRequest, clusterVersionPredicates).
+		Complete(c)
+}
+
+//+kubebuilder:rbac:groups=config.openshift.io,resources=clusterversions,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=config.openshift.io,resources=clusterversions/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=config.openshift.io,resources=clusterversions/finalizers,verbs=update
+//+kubebuilder:rbac:groups="apps",resources=deployments,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups="apps",resources=deployments/finalizers,verbs=update
+//+kubebuilder:rbac:groups="apps",resources=daemonsets,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups="apps",resources=daemonsets/finalizers,verbs=update
+//+kubebuilder:rbac:groups="storage.k8s.io",resources=csidrivers,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list;watch;create;update;delete
+//+kubebuilder:rbac:groups=security.openshift.io,resources=securitycontextconstraints,verbs=get;list;watch;create;patch;update
+//+kubebuilder:rbac:groups=monitoring.coreos.com,resources=prometheusrules,verbs=get;list;watch;create;update
+//+kubebuilder:rbac:groups="",resources=services,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=console.openshift.io,resources=consoleplugins,verbs=*
+
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
+func (c *OperatorConfigMapReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	var err error
+	c.ctx = ctx
+	c.log = log.FromContext(ctx, "OperatorConfigMap", req)
+	c.log.Info("Reconciling OperatorConfigMap")
+
+	c.operatorConfigMap = &corev1.ConfigMap{}
+	if err = c.Client.Get(context.TODO(), req.NamespacedName, c.operatorConfigMap); err != nil {
```

Comment:
Why `context.TODO` and not `c.ctx`?

Reply:
was actually a leftover from the previous logic that was here, fixed.